pilot/cmd/pilot-agent/main.go

```diff
@@ -67,15 +69,18 @@ const (
 // TODO: Move most of this to pkg.
 
 var (
-	role               = &model.Proxy{}
-	proxyIP            string
-	registryID         serviceregistry.ProviderID
-	trustDomain        string
-	pilotIdentity      string
-	mixerIdentity      string
-	statusPort         uint16
-	stsPort            int
-	tokenManagerPlugin string
+	role                        = &model.Proxy{}
+	proxyIP                     string
+	registryID                  serviceregistry.ProviderID
+	trustDomain                 string
+	pilotIdentity               string
+	mixerIdentity               string
+	statusPort                  uint16
+	stsPort                     int
+	tokenManagerPlugin          string
+	tlsServerDNSServerCertChain string
```

Comment:
See my comments - I would move this code to secretcache.go - in GenerateSecret or one of the related methods.